### PR TITLE
Extract tarballs in Ubuntu:14.04

### PR DESCRIPTION
**Pull request by PrajwalM2212**
Some tarballs cannot be extracted without running as root. Hence tar will be extracted by running the root command

Resolves https://github.com/vmware/tern/issues/433

Signed-off-by: Prajwal M <prajwalmmath@gmail.com>

### CODE REVIEW

File: tern/utils/rootfs.py [hunk 1]
```diff
@@ -86,8 +86,8 @@ def set_up():
 def extract_layer_tar(layer_tar_path, directory_path):
     '''Assuming all the metadata for an image has been extracted into the
     temp folder, extract the tarfile into the required directory'''
-    with tarfile.open(layer_tar_path) as tar:
-        tar.extractall(directory_path)
+    os.makedirs(directory_path)
+    root_command(extract_tar, layer_tar_path, '-C', directory_path)
```
Comment (nishakm) [inline tern/utils/rootfs.py]: Good catch! The trouble with shelling out to the system's tar utility is that it won't check for empty tarballs as we have seen in #430. I wonder if we can use `tarfile` to see if the tarball is empty or not before shelling out to it. If you can do this, I can close #226 as this change will resolve that one as well.

**Comment by nishakm** [comment]
@PrajwalM2212 on line 61
``` 
else:
        return result
```
Can you remove the `else:` and just `return result`?

File: tern/utils/rootfs.py (hunk 1 above)
Comment (PrajwalM2212) [reply tern/utils/rootfs.py]: Yes. I will check for empty tarballs.

Comment (PrajwalM2212) [reply tern/utils/rootfs.py]: @nishakm tarfile seems to not help. But `tar -tf` will help in this case. 
But bandit is throwing some security issues. Any clue on how to fix them ?

Comment (PrajwalM2212) [reply tern/utils/rootfs.py]: @nishakm `tarfile` module is not helping. `tar -tf` helps.  But there are some security warnings associated with this. Any clue how to resolve them ?

If they can't be resolved then we may have to find another way to validate tar files.

Comment (nishakm) [reply tern/utils/rootfs.py]: @PrajwalM2212 do you get any warnings when running `tar -tf` on an ubuntu image? I don't see any.

File: tern/utils/rootfs.py [hunk 2]
```diff
@@ -86,8 +85,12 @@ def set_up():
 def extract_layer_tar(layer_tar_path, directory_path):
     '''Assuming all the metadata for an image has been extracted into the
     temp folder, extract the tarfile into the required directory'''
-    with tarfile.open(layer_tar_path) as tar:
-        tar.extractall(directory_path)
+    os.makedirs(directory_path)
+    with open(os.devnull, 'w') as test:
+        result = subprocess.call(['tar', '-tf', layer_tar_path],
```
Comment (nishakm) [inline tern/utils/rootfs.py]: Oh! This is what you meant by security warning. You can run `root_command` here as well and check to see if there is an error or if the result is an empty string.

Comment (nishakm) [reply tern/utils/rootfs.py]: `root_command` will take care of the pipe redirection for you and will return the result and an error message.

**Comment by nishakm** [comment]
Closing this as the commit is in the PR #440

**Pull request closed without merging**